public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx

```diff
@@ -127,3 +134,14 @@ export const GeneralSettings: React.FC<Props> = ({ dashboard }) => {
     </div>
   );
 };
+
+const mapDispatchToProps = {
+  updateTimeZone: (timeZone: TimeZone) => {
+    updateTimeZoneForSession(timeZone);
+    getTimeSrv().refreshDashboard();
+  },
```

Comment:
I think we should create a proper ThunkAction in a `actions.ts` file which is our convention.

Reply:
good suggestion @hugohaggmark 🚀  787291e092d04907ebbc3db2a5824b8514f72af0